src/cls/IPM/Storage/Module.cls

```diff
@@ -1022,7 +1034,11 @@ Method GetResolvedReferences(Output pReferenceArray, pLockedDependencies As %Boo
 			
 			kill tempArray
 			Set tSC = tResource.ResolveChildren(.tempArray)
-			Merge pReferenceArray(..Name) = tempArray
+			Set resourceScope = tResource.Scope
+			Set attrScope = $Get(tempArray("Scope"))
+			If ..HasScope(pPhases, resourceScope) && ..HasScope(pPhases, attrScope) {
```

Comment:
Nit: the first check if already done above so only the second check needs to be done here